lib/pbench/agent/tool_data_sink.py

```diff
@@ -154,6 +156,31 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.volume = self.tool_group_dir / "prometheus"
 
+    def __create_yml(self, template, context):
```

Comment:
A more specific name would be hand here; also this is maybe an `_` internal method but not really an `__` method.

This is almost a generic "apply Jinja template" method; there's certainly no inherent dependency here on it being a yml file. Maybe this should be elevated to a global util method; but at the very least a name implying something more like "expand Jinja template" would be more informative.

Plus, a description string particularly to explain the "context".